datafusion/core/src/execution/session_state.rs

```diff
@@ -139,6 +139,8 @@ pub struct SessionState {
     analyzer: Analyzer,
     /// Provides support for customizing the SQL planner, e.g. to add support for custom operators like `->>` or `?`
     expr_planners: Vec<Arc<dyn ExprPlanner>>,
+    #[cfg(feature = "sql")]
```

Comment:
I like how this follows the same model as expr planners and type planners

I do think it might be worth considering (as a follow on PR) moving all the SQL related extensions into their own struct (as we now have type_planners, expr_planners and relation_planners)